docs/data/data-grid/editing/LinkedFieldsRowEditing.js

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Box from '@mui/material/Box';
+import {
+  DataGrid,
+  GridEditSingleSelectCell,
+  useGridApiContext,
+} from '@mui/x-data-grid';
+import { randomPrice } from '@mui/x-data-grid-generator';
+
+const rows = [
+  {
+    id: 1,
+    description: 'Light bill',
+    value: randomPrice(0, 1000),
+    type: 'Expense',
+    account: 'Utilities',
+  },
+  {
+    id: 3,
+    description: 'Order #5',
+    value: randomPrice(0, 1000),
+    type: 'Income',
+    account: 'Sales',
+  },
+  {
+    id: 4,
+    description: 'Google AdSense',
+    value: randomPrice(0, 1000),
+    type: 'Income',
+    account: 'Ads',
+  },
+];
+
+const CustomTypeEditComponent = (props) => {
+  const apiRef = useGridApiContext();
+
+  const handleChange = async () => {
+    await apiRef.current.setEditCellValue({
+      id: props.id,
+      field: 'account',
+      value: '',
+    });
+  };
+
+  return <GridEditSingleSelectCell onChange={handleChange} {...props} />;
+};
+
+CustomTypeEditComponent.propTypes = {
+  /**
+   * The grid row id.
+   */
+  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+};
+
+export default function LinkedFieldsRowEditing() {
+  const columns = [
+    { field: 'description', headerName: 'Description', width: 160, editable: true },
+    {
+      field: 'value',
+      headerName: 'Value',
+      type: 'number',
+      width: 120,
+      editable: true,
+    },
+    {
+      field: 'type',
+      headerName: 'Type',
+      type: 'singleSelect',
+      valueOptions: ['Income', 'Expense'],
+      width: 120,
+      editable: true,
+      renderEditCell: (params) => <CustomTypeEditComponent {...params} />,
+    },
+    {
+      field: 'account',
+      headerName: 'Account',
+      type: 'singleSelect',
+      valueOptions: ({ row }) => {
+        if (!row) {
+          return [
+            'Sales',
+            'Investments',
+            'Ads',
+            'Taxes',
+            'Payroll',
+            'Utilities',
+            'Marketing',
+          ];
+        }
+
+        return row.type === 'Income'
+          ? ['Sales', 'Investments', 'Ads']
+          : ['Taxes', 'Payroll', 'Utilities', 'Marketing'];
+      },
+      width: 140,
+      editable: true,
+    },
+  ];
+
+  return (
+    <Box sx={{ width: 1, height: 300 }}>
+      <DataGrid
+        rows={rows}
+        columns={columns}
+        editMode="row"
```

Comment:
I was surprised to see that this demo doesn't work with `editMode="cell"`.


https://user-images.githubusercontent.com/13808724/166456358-2d94b192-ff00-41d1-bada-4eeb710de473.mov

`setEditCellValue` returns a promise that never resolves if cell is not in edit mode.

https://github.com/mui/mui-x/blob/85ddb51a2160204ba2179821521074579c912407/packages/grid/x-data-grid/src/hooks/features/editRows/useGridEditing.new.ts#L129-L131

This seems wrong, should it just do nothing and resolve instead?

Reply:
To make it work with `editMode="cell"` then `apiRef.current.updateRows` needs to be used, because cell editing is to only update a single cell value.

> This seems wrong, should it just do nothing and resolve instead?

This was done on purpose. You have to consider that this call might have been debounced, so when it's called the cell may not be in edit mode anymore. If it resolves then means that it was able to update the state, but if cell is not in edit mode there's no state to update.

Another thing to consider is that the edit component may have the ["auto-stop"](https://mui.com/x/react-data-grid/editing/#with-auto-stop) behavior. In this way, `stopCellEditMode` is called after `setEditCellValue` resolves. Since `stopCellEditMode` throws an error if not edit mode, never resolving prevents a crash.

https://github.com/mui/mui-x/blob/85ddb51a2160204ba2179821521074579c912407/packages/grid/x-data-grid/src/hooks/features/editRows/useGridCellEditing.new.ts#L326-L330

Reply:
Ah right, in `editMode="cell"` there's no need to call `setEditCellValue` on `account` cell, since it's in `view` state and doesn't have an edit value.
I just tried `updateRows` and it does the job 🎉

I think we should have demos for both cell and row editing for linked fields, since the approach is different depending on editing mode.

Reply:
Where do we put it in the docs, since this demo is inside "Row editing"? The example for cell editing is a bit more complex than only calling `updateRows`. It also needs to handle when the user cancels the edit, where it should also revert the updated value. Actually, `onChange` is not even necessary for this as `processRowUpdate` could be used.

Reply:
I created an "Advanced use cases" section and added both demos there: https://deploy-preview-4621--material-ui-x.netlify.app/x/react-data-grid/editing/#advanced-use-cases

Reply:
Looks great!